src/Features/Core/Portable/Snippets/AbstractConsoleSnippetProvider.cs

```diff
@@ -0,0 +1,125 @@
+            var typeExpression = generator.TypeExpression(symbol!);
+            var declaration = GetAsyncSupportingDeclaration(token);
+            var isAsync = declaration is not null && generator.GetModifiers(declaration).IsAsync;
+
+            var invocation = isAsync
+                ? generator.ExpressionStatement(generator.AwaitExpression(generator.InvocationExpression(
+                    generator.MemberAccessExpression(generator.MemberAccessExpression(typeExpression, generator.IdentifierName(nameof(Console.Out))), generator.IdentifierName(nameof(Console.Out.WriteLineAsync))))))
+                : generator.ExpressionStatement(generator.InvocationExpression(generator.MemberAccessExpression(typeExpression, generator.IdentifierName(nameof(Console.WriteLine)))));
+            return new TextChange(TextSpan.FromBounds(position, position), invocation.NormalizeWhitespace().ToFullString());
+        }
+
+        protected override int? GetTargetCaretPosition(ISyntaxFactsService syntaxFacts, SyntaxNode caretTarget)
+        {
+            var invocationExpression = caretTarget.DescendantNodes().Where(syntaxFacts.IsInvocationExpression).FirstOrDefault();
+            if (invocationExpression is null)
+            {
+                return null;
+            }
+
+            var argumentListNode = syntaxFacts.GetArgumentListOfInvocationExpression(invocationExpression);
+
+            if (argumentListNode is null)
+            {
+                return null;
+            }
+
+            return argumentListNode!.GetLocation().SourceSpan.End - 1;
+        }
+
+        protected override async Task<SyntaxNode> AnnotateNodesToReformatAsync(Document document,
+            SyntaxAnnotation findSnippetAnnotation, SyntaxAnnotation cursorAnnotation, int position, CancellationToken cancellationToken)
+        {
+            var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
+            var snippetExpressionNode = GetConsoleExpressionStatement(syntaxFacts, root, position);
+            if (snippetExpressionNode is null)
+            {
+                return root;
+            }
+
+            var symbol = await GetSymbolFromMetaDataNameAsync(document, cancellationToken).ConfigureAwait(false);
+
+            var reformatSnippetNode = snippetExpressionNode.WithAdditionalAnnotations(findSnippetAnnotation, cursorAnnotation, Simplifier.Annotation, SymbolAnnotation.Create(symbol!), Formatter.Annotation);
+            return root.ReplaceNode(snippetExpressionNode, reformatSnippetNode);
+        }
+
+        private static SyntaxNode? GetConsoleExpressionStatement(ISyntaxFactsService syntaxFacts, SyntaxNode root, int position)
+        {
+            var closestNode = root.FindNode(TextSpan.FromBounds(position, position));
+            return closestNode.FirstAncestorOrSelf<SyntaxNode>(syntaxFacts.IsExpressionStatement);
```

Comment:
you still need to check that the nearest expression statement starts on the position you expect.